### PR TITLE
[Infra] Add git diff to repo manager class

### PR DESCRIPTION
**Pull request by Leo-Neat**
This adds the functionality of git diff --name-only to the repo manager module. This functionality will be used for the affected fuzzer feature of CIFuzz. Associated tests have also been added.

### CODE REVIEW

File: infra/repo_manager.py [hunk 1]
```diff
@@ -93,6 +94,24 @@ def commit_exists(self, commit):
                                    self.repo_dir)
     return not err_code
 
+  def get_git_dif(self):
```
Comment (jonathanmetzman) [inline infra/repo_manager.py]: "diff"
Why this is a standalone function

File: infra/repo_manager_test.py [hunk 2]
```diff
@@ -133,5 +135,59 @@ def test_checkout_invalid_pull_request(self):
         test_repo_manager.checkout_pr('not/a/valid/pr')
 
 
+class GitDiffUnitTests(unittest.TestCase):
```
Comment (jonathanmetzman) [inline infra/repo_manager_test.py]: Don't we normally name these singular, e.g. `GitDiffUnitTest` same for next class.

File: infra/repo_manager.py [hunk 3]
```diff
@@ -93,6 +94,24 @@ def commit_exists(self, commit):
...
+    self.fetch_unshallow()
+    out, err_msg, err_code = utils.execute(
+        ['git', 'diff', '--name-only', 'origin'], self.repo_dir)
+    if err_code:
+      logging.error('Git diff failed with error message %s.', err_msg)
+      return None
+    if not out:
+      logging.info('No diff was found.')
+      return None
+    file_diff = list(filter(None, out.split('\n')))
```
[the remark this answers is not in the document]
Comment (Leo-Neat) [reply infra/repo_manager.py]: Done.

File: infra/repo_manager_test.py [hunk 4]
```diff
@@ -15,9 +15,11 @@
 
 import os
 import unittest
+import unittest.mock
```
[the remark this answers is not in the document]
Comment (Leo-Neat) [reply infra/repo_manager_test.py]: Added.

File: infra/repo_manager_test.py [hunk 5]
```diff
@@ -118,7 +120,7 @@ def test_checkout_valid_pull_request(self):
       test_repo_manager = repo_manager.RepoManager(OSS_FUZZ_REPO, tmp_dir)
       test_repo_manager.checkout_pr('refs/pull/3415/merge')
       self.assertEqual(test_repo_manager.get_current_commit(),
-                       '314c9249a54a08e764a5bbcb7333294ae7c1f9ed')
+                       '2d9759999071e2c0843c63129d8426282a1a7669')
```
[the remark this answers is not in the document]
Comment (Leo-Neat) [reply infra/repo_manager_test.py]: I did not catch it during the test PR that I submitted last week. This should be final.

File: infra/repo_manager_test.py (hunk 2 above)
Comment (Leo-Neat) [reply infra/repo_manager_test.py]: Done.

File: infra/repo_manager.py [hunk 6]
```diff
@@ -93,6 +94,23 @@ def commit_exists(self, commit):
...
+      A list of changed file paths or None on Error.
+    """
+    self.fetch_unshallow()
+    out, err_msg, err_code = utils.execute(
+        ['git', 'diff', '--name-only', 'origin'], self.repo_dir)
+    if err_code:
+      logging.error('Git diff failed with error message %s.', err_msg)
+      return None
+    if not out:
+      logging.info('No diff was found.')
```
Comment (Dor1s) [inline infra/repo_manager.py]: shouldn't this be an error too?

File: infra/repo_manager_test.py [hunk 7]
```diff
@@ -109,16 +111,55 @@ def test_invalid_commit_list(self):
...
+  def test_diff_no_change(self):
+    """Tests that None is returned when there is no difference between repos."""
+    with tempfile.TemporaryDirectory() as tmp_dir:
+      repo_man = repo_manager.RepoManager(OSS_FUZZ_REPO, tmp_dir)
+      diff = repo_man.get_git_diff()
+      self.assertIsNone(diff)
+
+
+class CheckoutPRIntegrationTest(unittest.TestCase):
+  """Class testing functionality of get_git_diff in the repo_manager module."""
```
Comment (Dor1s) [inline infra/repo_manager_test.py]: I guess you meant `Class to test the functionality of checkout_pr of the RepoManager class` from lien 113 on the left :)

Comment (Leo-Neat) [reply infra/repo_manager_test.py]: Fixed.

File: infra/repo_manager.py (hunk 6 above)
Comment (Leo-Neat) [reply infra/repo_manager.py]: Fixed.

**Pull request merged**